gsdk/src/api.rs

```diff
@@ -37,8 +37,8 @@ pub struct Api {
 
 impl Api {
     /// Create new API client.
-    pub async fn new(url: impl AsOption<str>) -> Result<Self> {
-        Self::new_with_timeout(url, None).await
+    pub async fn new(url: impl Into<Option<String>>) -> Result<Self> {
```

Comment:
&str as well

Reply:
mb `AsRef<str>`?

Reply:
the problem here is `String` is most frequently used for this interface ...

```
// impl Into<Option<String>>
let api = Api::new(node.ws());

// impl Into<Option<&str>>
let api = Api::new(node.ws().as_str())

// impl Into<Option<T>> where T: AsRef<str>
let api = Api::new::<String>(node.ws())
```

Reply:
there problem here is, `String` is the most frequently used case

```rust
// impl Into<Option<String>>
let api = Api::new(node.ws());
let api = Api::new(Some("".to_string()))
let api = Api::new(None)

// impl Into<Option<&str>>
let api = Api::new(node.ws().as_str())
let api = Api::new(Some(""))
let api = Api::new(None)

// impl<T: AsRef<str>> Into<Option<T>>
let api = Api::new(node.ws())
let api = Api::new::<String>(Some("".to_string()))
let api = Api::new::<String>(None)
```

Reply:
makes sense if rust cannot infer type